docs/source/miscellaneous/doc/adr/0017-transactions3.md

```diff
@@ -0,0 +1,302 @@
+One can simply set the `targetTransactionsSchemaVersion` in their configuration to `3`, and we will (after going 
+through the required schema lease process) use _transactions3.
+
+We have set up our existing transaction tests to become parameterised, and have run them with both legacy and V3
+transaction schemas. Curiously, at the time _transactions2 was written these were *not* wired up. We have
+also implemented a simulated test version of the consensus forgetting store that can forget values with configurable
+probability, and have used this to fuzz test the put-unless-exists table to give us stronger evidence that the protocol
+is safe, and that it is implemented correctly.
+
+## Alternatives Considered
+
+- One possible solution could be to perform reads at SERIAL consistency, and override the default system keyspace TTL in 
+  Cassandra to be longer. However, we would still need to pick a reasonable bound after which the Paxos logs and other
+  contents of the system keyspace could be kept for. Furthermore, reads at SERIAL consistency will not achieve eventual
+  read efficiency; these require Cassandra nodes to perform a round of Paxos internally, adding two *more* RPCs between
+  the coordinator (which acts as a Paxos proposer) and the other nodes.
+- Another possible approach involves a state machine where a user is allowed to propose their own value as part of the
+  read protocol (as opposed to being required to commit the staging value that was written). The author thinks this
+  could probably work, but it is more difficult to reason about (e.g. one has to be more concerned about the behaviour
+  of concurrent proposals, and whether a dueling proposers-style situation is possible).
+
+## Consequences
+As transactions3 is rolled out globally to Cassandra deployments, they will no longer be exposed to this correctness
+bug. We hope (though don't have strong evidence owing to small sample sizes and extreme difficulty in root causing) 
+that this will reduce the incidence of AtlasDB corruption tickets in general.
+
+Backup and restore workflows, or any use cases that manually manipulate one of the transaction tables will need to be
+aware that new serialized forms exist. In particular, the transactions2 table should not be read in isolation without
+caution - while values are unambiguous in indicating whether they are written with transactions2 or transactions3,
+a user needs to be aware that a STAGING value is not safe to use as though it is final.
+
+It is possible, even likely, that the performance of transactions3 is worse than that of transactions2 (we do strictly
+more work, and values that need to be passed to and from the database are strictly larger). However, we don't expect
+this to add more than a constant amount of overhead, and this should be very small in the steady state. This isn't a
+very fair comparison, in any case, as the transactions2 approach is not correct.
+
+## Appendices
```

Comment:
nit:
```suggestion
## Appendix
```
(there's only one)

Reply:
yeah. I think I originally was going to put all the long proofs here but turns out there was just one.